tests/apps/injectorapp/app.go

```diff
@@ -128,7 +161,28 @@ func appRouter() *mux.Router {
 	return router
 }
 
+func securedAppRouter() *mux.Router {
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Log requests and their processing time
+	router.Use(utils.LoggerMiddleware)
+
+	router.HandleFunc("/", indexHandler).Methods("GET")
+
+	router.Use(mux.CORSMethodMiddleware(router))
+
+	return router
+}
+
 func main() {
 	log.Printf("Injector App - listening on http://localhost:%d", appPort)
-	utils.StartServer(appPort, appRouter, true)
+
+	// start a secured app (with TLS) on an alternate port
+	go func() {
```

Comment:
Should this be a different e2e test app instead? Also for port forwarding reasons?

Reply:
Perhaps could re-use the app "binding-output"?

Reply:
Yes, I think it's best to move it to a new app to avoid confusions. Thank you!

Reply:
There seems to be one issue here - injectorapp's init container is generating the SSL certs that is (1) used by the web server to listen to TLS, and (2) used by the Dapr sidecar of the test app to invoke the bindings with.

Everything comes together here:
https://github.com/dapr/dapr/blob/6bf103ec83b6fb92389d717a5f38c063137414c4/tests/e2e/injector/injector_test.go#L81-L107

If I move the cert generation logic and web server to another app, I will need a way to share those with the test app. Do you have any potential ideas on that? One way can be secrets, but I need to explore how to create them from the init container.

Reply:
I see, that makes sense. I think your current approach is perhaps the best one then!

Reply:
Thank you, removing the WIP from the PR.